microservices-connector/cmd/router/main.go

```diff
@@ -394,20 +399,24 @@ func handleSequencePipeline(nodeName string,
 	return responseBytes, statusCode, nil
 }
 
-func routeStep(nodeName string, graph mcv1alpha3.GMConnector, input []byte, headers http.Header) ([]byte, int, error) {
+func routeStep(nodeName string,
+	graph mcv1alpha3.GMConnector,
+	initInput, input []byte,
+	headers http.Header,
+) ([]byte, int, error) {
 	defer timeTrack(time.Now(), "node", nodeName)
 	currentNode := graph.Spec.Nodes[nodeName]
 
 	if currentNode.RouterType == mcv1alpha3.Switch {
-		return handleSwitchPipeline(nodeName, graph, input, headers)
+		return handleSwitchPipeline(nodeName, graph, initInput, input, headers)
```

Comment:
since we have `headers` for the incoming request, can we use it instead of add `initInput` ?  the reason is there are cases we will need new 'input` + original user request headers instead of the `initInput`

Reply:
Hi @irisdingbj, I have talked with MI6 team, they pass their parameters via request body instead of request header, so I have to use the initInput.